### ISSUE TITLE
test_sparse_hammerblade_empty_path2 FAILED on COSIM

### ISSUE THREAD

**Issue by yodada** [cornell-brg/hb-pytorch#25]
## 🐛 Bug

```
test_sparse_hammerblade_empty_path2 FAILED [ 65%]
```

```
>       assert torch.equal(hb_v.cpu(), cpu_v)
E       AssertionError: assert False
E        +  where False = <built-in method equal of type object at 0x2aaac987e420>(tensor([1.5408e-37, 1.7195e-37,        nan, 4.4721e+21, 2.6769e+20, 4.5443e+30,\n        1.6633e+22, 9.5835e+26, 4.1984...4, 2.2421e-44, 2.1019e-44, 1.9618e-44, 1.8217e-44, 1.6816e-44,\n        1.5414e-44, 1.4013e-44, 1.2612e-44, 1.1210e-44]), tensor([1.5408e-37, 1.7195e-37,        nan, 4.4721e+21, 2.6769e+20, 4.5443e+30,\n        1.6633e+22, 9.5835e+26, 4.1984...4, 2.2421e-44, 2.1019e-44, 1.9618e-44, 1.8217e-44, 1.6816e-44,\n        1.5414e-44, 1.4013e-44, 1.2612e-44, 1.1210e-44]))
E        +    where <built-in method equal of type object at 0x2aaac987e420> = <module 'torch' from '/work/global/lc873/work/sdh/venv-hb-device/lib64/python3.6/site-packages/torch/__init__.py'>.equal
E        +    and   tensor([1.5408e-37, 1.7195e-37,        nan, 4.4721e+21, 2.6769e+20, 4.5443e+30,\n        1.6633e+22, 9.5835e+26, 4.1984...4, 2.2421e-44, 2.1019e-44, 1.9618e-44, 1.8217e-44, 1.6816e-44,\n        1.5414e-44, 1.4013e-44, 1.2612e-44, 1.1210e-44]) = <built-in method cpu of Tensor object at 0x2aaacf0baf78>()
E        +      where <built-in method cpu of Tensor object at 0x2aaacf0baf78> = tensor([1.5408e-37, 1.7195e-37,        nan, 4.4721e+21, 2.6769e+20, 4.5443e+30,\n        1.6633e+22, 9.5835e+26, 4.1984...-44, 1.9618e-44, 1.8217e-44, 1.6816e-44,\n        1.5414e-44, 1.4013e-44, 1.2612e-44, 1.1210e-44], device='hammerblade').cpu

tests/test_hammerblade_sparse_backend.py:70: AssertionError
```

Seems to be a floating point comparison issue?

**Comment by yodada** [comment]
https://github.com/cornell-brg/hb-pytorch/pull/47#issuecomment-608908433

**Comment by yodada** [comment]
actually, I think another way to fix this is to not use `empty` but `randn` for tensor creation.
https://github.com/cornell-brg/hb-pytorch/blob/b64c0c07fc9b43e89cea34a0e9c09baa40638f0f/hammerblade/torch/tests/test_hammerblade_sparse_backend.py#L30

**Comment by yodada** [comment]
Fixed by #48
